core/arch/arm/plat-rockchip/platform.c

```diff
@@ -33,9 +34,35 @@ static TEE_Result platform_init(void)
 	io_write32(sgrf_base + SGRF_SOC_CON(9), SLAVE_ALL_NS);
 	io_write32(sgrf_base + SGRF_SOC_CON(10), SLAVE_ALL_NS);
 
-	return TEE_SUCCESS;
+	return 0;
+}
+
+static int platform_secure_ddr_region(int rgn, paddr_t st, size_t sz)
+{
+	MSG("not protecting region %d: 0x%lx-0x%lx\n", rgn, st, st + sz);
+
+	return 0;
 }
 
 #endif
 
+static TEE_Result platform_init(void)
+{
+	int ret;
+
+	platform_secure_init();
```

Comment:
Minor: since the intent is to have platform-specific variants of `platform_secure_init()` and `platform_secure_ddr_region()`, I would personally add a header file: `platform.h`, included via `#include "platform.h"`, and move the implementation of these two functions to `platform_rk322x.c`. This way we don't need any ifdefery in `platform.c` and no further change to this file when adding a new platform flavor -- only one new C file and an update to `sub.mk`.

But, I admit this is almost a personal taste thing so you may as well keep the code as it is.

Reply:
I don't really have a hard preference here :-) ... But some if-defery will always need to happen. It's either in platform.c or in sub.mk.

Reply:
> But some if-defery will always need to happen. It's either in platform.c or in sub.mk.

Agreed. My point was, I prefer the conditionals in the makefiles than in `.c` files. In this case it wouldn't even look like a conditional in `sub.mk`, it would be: `src-$(PLATFORM_FLAVOR_rk322x) += platform_rk322x.c`. For files used by several flavors that would be an `ifeq ()...endif` however.